apps/wallet/src/ui/app/pages/home/nft-transfer/TransferNFTForm.tsx

```diff
@@ -1,122 +1,147 @@
-import { useEffect, useRef, memo } from 'react';
+import { ArrowRight16 } from '@mysten/icons';
+import { getTransactionDigest, SUI_TYPE_ARG } from '@mysten/sui.js';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { Form, Field, Formik } from 'formik';
+import { toast } from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
 
-import { Content } from '_app/shared/bottom-menu-layout';
-import Button from '_app/shared/button';
+import { createValidationSchema } from './validation';
+import { useActiveAddress } from '_app/hooks/useActiveAddress';
+import { Button } from '_app/shared/ButtonUI';
+import BottomMenuLayout, {
+    Content,
+    Menu,
+} from '_app/shared/bottom-menu-layout';
+import { Text } from '_app/shared/text';
 import { AddressInput } from '_components/address-input';
 import Alert from '_components/alert';
-import Icon, { SuiIcons } from '_components/icon';
-import LoadingIndicator from '_components/loading/LoadingIndicator';
-import { useGetCoinBalance, useAppSelector } from '_hooks';
+import Loading from '_components/loading';
+import { useGetCoinBalance, useSigner } from '_hooks';
+import { DEFAULT_NFT_TRANSFER_GAS_FEE } from '_redux/slices/sui-objects/Coin';
 import { useGasBudgetInMist } from '_src/ui/app/hooks/useGasBudgetInMist';
 
-import type { FormValues } from '.';
-
-import st from './TransferNFTForm.module.scss';
-
-export type TransferNFTFormProps = {
-    submitError: string | null;
-    gasBudget: number;
-    onClearSubmitError: () => void;
-};
-
-function TransferNFTForm({
-    submitError,
-    gasBudget,
-    onClearSubmitError,
-}: TransferNFTFormProps) {
-    const {
-        isSubmitting,
-        isValid,
-        values: { to },
-    } = useFormikContext<FormValues>();
-    const onClearRef = useRef(onClearSubmitError);
-    onClearRef.current = onClearSubmitError;
-    useEffect(() => {
-        onClearRef.current();
-    }, [to]);
-
-    const accountAddress = useAppSelector(({ account }) => account.address);
+export function TransferNFTForm({ objectId }: { objectId: string }) {
+    const activeAddress = useActiveAddress();
+    const validationSchema = createValidationSchema(
+        activeAddress || '',
+        objectId
+    );
     const { data: coinBalance, isLoading: loadingBalances } = useGetCoinBalance(
         SUI_TYPE_ARG,
-        accountAddress
+        activeAddress
     );
+    const signer = useSigner();
+    const queryClient = useQueryClient();
+    const navigate = useNavigate();
+    const transferNFT = useMutation({
+        mutationFn: (to: string) => {
+            if (!to || !signer || isInsufficientGas) {
+                throw new Error('Missing data');
+            }
+            return signer.transferObject({
+                recipient: to,
+                objectId,
+                gasBudget: DEFAULT_NFT_TRANSFER_GAS_FEE,
+            });
+        },
+        onSuccess: async (response) => {
+            Promise.all([
```

Comment:
What's the expectation here? Why we use promise all? Do we need to await as well? Do we care if any of the promises fail?

Reply:
see https://github.com/TanStack/query/discussions/1738